listenbrainz/webserver/static/js/src/recommendations/Recommendations.tsx

```diff
@@ -62,6 +65,66 @@ export default class Recommendations extends React.Component<
     );
   }
 
+  componentDidMount(): void {
+    const { user, currentUser } = this.props;
+      if (currentUser?.name === user?.name) {
+        this.loadFeedback();
+      }
+  }
+
+  getFeedback = async () => {
+    const { user, recommendations } = this.props;
+    let recordings = "";
+
+    if (recommendations) {
+      recommendations.forEach((recommendation) => {
+        const recordingMbid = get(
+          recommendation,
+          "track_metadata.additional_info.recording_mbid"
+        );
+        if (recordingMbid) {
+          recordings += `${recordingMbid},`;
```

Comment:
but why do we want to declare an array and not an empty string?

Reply:
`Variable 'recordings' implicitly has type 'any[]' in some locations where its type cannot be determined.` is what I get. maybe because APIService.getFeedbackForUserForRecommendations expect recordings to be of type string

Reply:
`Argument of type 'any[]' is not assignable to parameter of type 'string'.ts(2345)
Variable 'recordings' implicitly has an 'any[]' type.`

Reply:
This works for me:
```
getFeedback = async () => {
    const { user, recommendations } = this.props;
    const recordings: string[] = [];

    if (recommendations) {
      recommendations.forEach((recommendation) => {
        const recordingMbid = get(
          recommendation,
          "track_metadata.additional_info.recording_mbid"
        );
        if (recordingMbid) {
          recordings.push(recordingMbid);
        }
      });
      try {
        const data = await this.APIService.getFeedbackForUserForRecommendations(
          user.name,
          recordings.join(",")
        );
        return data.feedback;
      } catch (error) {
        this.newAlert(
          "danger",
          "Playback error",
          typeof error === "object" ? error.message : error
        );
      }
    }
    return {};
  };
```

The three changes are:
- `const recordings: string[] = [];` we defined an empty array of strings with the proper type
- `recordings.push(recordingMbid);`  we're pushing the MBID to the array, faster and easier to manipulate than concatenating to a string. If we needed (or will in the future), we can then use array functions to modify or filter the array, which would be lot easier than if we had a string !
- `recordings.join(",")` we call this only once at the end to generate a string of MBIDs separated by commas (again, faster that modifying the string every time in the forEach loop.